flink-java/src/test/java/org/apache/flink/api/java/operator/SortPartitionTest.java

```diff
@@ -169,6 +169,38 @@ public void testSortPartitionWithExpressionKeys4() {
 		tupleDs.sortPartition("f3", Order.ASCENDING);
 	}
 
+	@Test
+	public void testSortPartitionWithKeySelector1() {
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		DataSet<Tuple4<Integer, Long, CustomType, Long[]>> tupleDs = env.fromCollection(tupleWithCustomData, tupleWithCustomInfo);
+
+		// should work
+		try {
+			tupleDs.sortPartition(new KeySelector<Tuple4<Integer, Long, CustomType, Long[]>, Integer>() {
+				@Override
+				public Integer getKey(Tuple4<Integer, Long, CustomType, Long[]> value) throws Exception {
+					return value.f0;
```

Comment:
The test only works because the KeySelector function returns the first field of the input data. If you change this to `value.f1` the test will fail.